### ISSUE TITLE
Difficulty changes

### ISSUE THREAD

**Issue by CitoplasmaX** [pooler/cpuminer#147]
Why are you closing my issues?, i can have more doubts.

I am openning these issues because i think that this software has a problem, on any pool with any machine or computer with this cpuminer and for any internet connection, difficulty changes take more time that on other mining software. 

Why could it happen?, i have not idea.

**Comment by pooler** [comment]
Sorry if I wasn't clear enough, maybe I should have been more explicit. I closed your tickets because, as I've already told you, the difficulty is set by the server (which, if you are using a pool, is controlled by the pool), not by the client (cpuminer). The protocol grants the server total freedom to change the difficulty on a given connection at any point in time, based on whatever criteria it deems fit. The server usually decides how to change the difficulty based on the estimated hash rate of the client, but nothing stops it from also considering other elements, such as what software the client is using, which day of the week it is, or whether there is a full moon. Except for non-standard practices that some pools have introduced to let the client *suggest* a difficulty to the server (but the server still decides), the client has absolutely no say in it. And, just to avoid misunderstandings, that is perfectly fine (as long as, of course, the server implementation is sensible).

You asked why it takes the server a long time to change the difficulty when you connect cpuminer to it. I cannot possibly know, since that's a server implementation detail, and that's why I suggested that you ask the operator of your server. Given that this is the bug tracker for cpuminer and that what you describe does not indicate a bug in cpuminer, I then closed the issue.
